superbench/benchmarks/micro_benchmarks/computation_communication_overlap.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Module of the ComputationCommunicationOverlap benchmarks.
+
+ComputationCommunicationOverlap benchmark is used to test the performance of single node
+when communication and computation overlap and figure out the issued GPU
+if the node has performance downgrade on this context:
+    -Currently, 2 computation kernels are supported: mul and matmul of the matrix
+    -Communication kernel: NCCL AllReduce
+    -Each GPU will run the computation kernel and communication kernels in pipeline
+        to achieve overlap in some degree, in every loop,
+        they will fist launch a nccl allreduce kernel async
+        and then launch #ratio computation kernels at once。
+"""
+
+import os
+import time
+
+# TODO - add mechanism to import torch as needed according to docker
+import torch
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmark
+from superbench.benchmarks.context import Enum
+
+
+class ComputationKernelType(Enum):
+    """The Enum class representing different computation kernel type."""
+    MATMUL = 'matmul'
+    MUL = 'mul'
+
+
+class ComputationCommunicationOverlap(MicroBenchmark):
+    """The base class of micro-benchmarks."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+        # Command lines to launch the micro-benchmarks.
```

Comment:
Please remove the member self.__commands, I will remove it in 'ShardingMatmul' too.